src/deadline/client/ui/dialogs/submit_job_progress_dialog.py

```diff
@@ -361,3 +349,113 @@ def _build_ui(self):
 
         self.layout.addWidget(self.progress_bar)
         self.layout.addWidget(self.progress_message)
+
+
+class ScrollableWarningDialog(QDialog):
+    """
+    Simple Dialog which functions similar to a QMessageBox, but with a scrollable text area.
+    """
+
+    def __init__(self, message: str, default_response: bool = False):
+        super().__init__()
+        self.setWindowTitle("Job Submission Confirmation")
+        self.message = message
+        self.default_response = default_response
+        self.buttons = None
+        layout = QVBoxLayout(self)
+
+        # Top section with icon and title
+        top_layout = QHBoxLayout()
+        icon_label = QLabel()
+        icon_label.setPixmap(
+            self.style()
+            .standardIcon(QStyle.SP_DirIcon if default_response else QStyle.SP_MessageBoxWarning)
+            .pixmap(32, 32)
+        )
+        top_layout.addWidget(icon_label)
+        title_label = QLabel("Job submission confirmation")
+        title_label.setStyleSheet("font-weight: bold;")
+        top_layout.addWidget(title_label)
+        top_layout.addStretch()
+        layout.addLayout(top_layout)
+
+        # Scrollable Text Area
+        self.text_edit = QTextEdit()
+        self.text_edit.setPlainText(message)
+        self.text_edit.setReadOnly(True)
+        layout.addWidget(self.text_edit)
+
+        # Set minimum width and calculate optimal height
+        self.text_edit.setMinimumWidth(500)
+        self._calculate_optimal_size()
+
+        self.buttons = QDialogButtonBox(QDialogButtonBox.Ok | QDialogButtonBox.Cancel)
+        self.buttons.accepted.connect(self.accept)
+        self.buttons.rejected.connect(self.reject)
+
+        if default_response:
+            # If the default response is to continue, add the "Do not ask again" button
+            dont_ask_button = QPushButton("Do not ask again", self)
+            dont_ask_button.clicked.connect(lambda: set_setting("settings.auto_accept", "true"))
+            dont_ask_button.clicked.connect(self.accept)
+            self.buttons.addButton(dont_ask_button, QDialogButtonBox.ActionRole)
+
+        layout.addWidget(self.buttons)
+
+    def _calculate_optimal_size(self):
+        """Calculate optimal height based on text content and font metrics."""
+        # Get font metrics for the text edit
+        font_metrics = QFontMetrics(self.text_edit.font())
+
+        # Calculate text width accounting for margins and scrollbar
+        text_width = self.text_edit.minimumWidth() - 40  # Account for margins and scrollbar
+
+        # Calculate the height needed for the text
+        text_rect = font_metrics.boundingRect(0, 0, text_width, 0, Qt.TextWordWrap, self.message)
+        text_height = text_rect.height()
+
+        # Add some padding for better appearance
+        padding = 20
+        optimal_height = text_height + padding
+
+        # Set reasonable bounds
+        min_height = 100
+        max_height = 500
```

Comment:
Should this be based on the screen size?

Those with bigger displays may want to be able to read as much of the warning as possible without needing to scroll. I doubt anyone uses a screen with less that 500 vertical pixels these days, but...